### ISSUE TITLE
Bug: Msgspec constraints ignored in MsgspecDTO

### ISSUE THREAD

**Issue by 9128305** [litestar-org/litestar#3026]
### Discussed in https://github.com/orgs/litestar-org/discussions/3025

<div type='discussions-op-text'>

<sup>Originally posted by **9128305** January 25, 2024</sup>
Msgpec constraint here is ignored when sending `{"foo": "t"}`. What am I doing wrong?

```python
from typing import Annotated

import msgspec
from litestar import post, Litestar
from litestar.dto import MsgspecDTO


class Request(msgspec.Struct):
    foo:  Annotated[str, msgspec.Meta(min_length=3,)]


@post("/example/", dto=MsgspecDTO[Request],sync_to_thread=False)
def example(data: Request) -> Request:
    return data


app = Litestar([example])

**Comment by peterschutt** [comment]
Thanks @9128305 - we should be copying these constraints onto the transfer models.

**Comment by 9128305** [comment]
Should be Annotated type annotation splitted to unwrapped and metadata? Or is viable only for msgspec? https://github.com/litestar-org/litestar/blob/6e4e530445eadbc1fd2f65bebca3bc68cf12f29a/litestar/typing.py#L495

**Comment by peterschutt** [comment]
> Should be Annotated type annotation splitted to unwrapped and metadata? Or is viable only for msgspec?
> 
> https://github.com/litestar-org/litestar/blob/6e4e530445eadbc1fd2f65bebca3bc68cf12f29a/litestar/typing.py#L495

Sorry, I don't understand the question, can you rephrase or give an example?
